sc_demographics.R

```diff
@@ -0,0 +1,280 @@
+    # use chi dob if avaliable
+    dob = if_else(is.na(chi_date_of_birth), submitted_date_of_birth, chi_date_of_birth)
+  )
+
+
+## postcode ##
+# clean-up
+sc_demog <- sc_demog %>%
+  mutate(
+    # remove any postcodes with length < 5
+    submitted_postcode = na_if(submitted_postcode, str_length(submitted_postcode) < 5 | str_length(submitted_postcode) == 5),
+    chi_postcode = na_if(chi_postcode, str_length(chi_postcode) < 5 | str_length(chi_postcode) == 5)
+  ) %>%
+  mutate(
+    # remove spaces
+    submitted_postcode = str_replace_all(submitted_postcode, fixed(" "), ""),
+    chi_postcode = str_replace_all(chi_postcode, fixed(" "), "")
+  ) %>%
+  mutate(
+    # remove postcodes with length 8 or longer
+    submitted_postcode = na_if(submitted_postcode, str_length(submitted_postcode) > 8 | str_length(submitted_postcode) == 8),
+    chi_postcode = na_if(chi_postcode, str_length(chi_postcode) > 8 | str_length(chi_postcode) == 8)
+  )
+
+sc_demog <-
+  sc_demog %>%
+  mutate(
+    # add space to create length 7 postcode
+    # length 5 postcode
+    # submitted postcode
+    submitted_postcode = if_else(str_length(submitted_postcode) == 5 & !is.na(submitted_postcode),
+      paste0(substr(submitted_postcode, 1, 2), " ", substr(submitted_postcode, 3, 5)),
+      submitted_postcode
+    ),
+    # chi postcode
+    chi_postcode = if_else(str_length(chi_postcode) == 5 & !is.na(chi_postcode),
+      paste0(substr(chi_postcode, 1, 2), " ", substr(chi_postcode, 3, 5)),
+      chi_postcode
+    )
+  ) %>%
+  mutate(
+    # length 6 postcode
+    # submitted postcode
+    submitted_postcode = if_else(str_length(submitted_postcode) == 6 & !is.na(submitted_postcode),
+      paste0(substr(submitted_postcode, 1, 3), " ", substr(submitted_postcode, 4, 6)),
+      submitted_postcode
+    ),
+    # chi postcode
+    chi_postcode = if_else(str_length(chi_postcode) == 6 & !is.na(chi_postcode),
+      paste0(substr(chi_postcode, 1, 3), " ", substr(chi_postcode, 4, 6)),
+      chi_postcode
+    )
+  )
+
+
+# invalid postcodes
+invalid_postcodes <- c(
+  "DY103DJ",
+  "EH191TR",
+  "EH292EZ",
+  "EH33TNZ",
+  "G46 2NF",
+  "G46 6FY",
+  "G69 2YB",
+  "G73 8NZ",
+  "G74 7SN",
+  "G75 1ZZ",
+  "G75 6DF",
+  "G77 3GT",
+  "G78 6BU",
+  "G78 ITE",
+  "G79 8AJ",
+  "IV178ED",
+  "KA113FP",
+  "KA28OBE",
+  "KA5 1LQ",
+  "KA71JUJ",
+  "KA9 9FG",
+  "KW152SE",
+  "KY1 3DO",
+  "KY13OPX",
+  "L15 0PR",
+  "M16 0GS",
+  "ML3 0GS",
+  "ML7 6AQ",
+  "NK1 1AA",
+  "NK1 1AB",
+  "PA438JP",
+  "PA494JS",
+  "PR2 5AL",
+  "TD8 8JD"
+)
+
+sc_demog <-
+  sc_demog %>%
+  mutate(
+    # remove dummy postcodes "NK1 0AA" and "NK1 1AB"
+    submitted_postcode = na_if(
+      submitted_postcode,
+      submitted_postcode == "NK1 0AA" | submitted_postcode == "NF1 1AB"
+    ),
+    chi_postcode = na_if(
+      chi_postcode,
+      chi_postcode == "NK1 0AA" | chi_postcode == "NF1 1AB"
+    )
+  ) %>%
```

Comment:
Great start 🙂
This comes up a lot but we can simplify this type of code with [dplyr::across](https://dplyr.tidyverse.org/reference/across.html)

```suggestion
  mutate(
    # remove dummy postcodes "NK1 0AA" and "NK1 1AB"
    across(c(submitted_postcode, chi_postcode),
               ~na_if(.x, .x== "NK1 0AA" | .x== "NF1 1AB")
               )
  ) %>%
```

Or even
```suggestion
  mutate(
    # remove dummy postcodes "NK1 0AA" and "NK1 1AB"
    across(ends_with("_postcode"),
               ~na_if(.x, .x== "NK1 0AA" | .x== "NF1 1AB")
               )
  ) %>%
```